c2/cuda/groth16_cuda.cu

```diff
@@ -28,6 +28,15 @@ typedef fr_t scalar_t;
 #include <msm/pippenger.cuh>
 #include <msm/pippenger.hpp>
 
+class groth16_error : public std::runtime_error {
+    int _code;
+public:
+    groth16_error(int err, const std::string& reason) : std::runtime_error{reason}
+    {   _code = err;   }
+    inline int code() const
+    {   return _code;   }
+};
```

Comment:
I'm renaming cuda_error to sppark_error in sppark/util/exception.cuh. So idea would be to drop this and use sppark_error for everything. CUDA_OK still works. I need to harmonize rust/src/lib.rs with the change...

Reply:
Done. Idea is to omit this class and switch to sppark_error instead. And omit multiple catch statement.